dependencies.yaml

```diff
@@ -17,31 +40,37 @@ dependencies:
     common:
       - output_types: [conda]
         packages:
-          - cmake>=3.24.0,!=3.30.0
-          - &legate_core legate-core=24.06.*
           - openblas
-          - python-build
+      - output_types: [conda, pyproject, requirements]
+        packages:
+          - cmake>=3.24.0,!=3.30.0
+          - &legate_core legate-core==24.06.*
           - scikit-build>=0.18.0
           - setuptools>=70.0
-  docs:
+  build_tools:
     common:
       - output_types: [conda]
+        packages:
+          - python-build>=1.2.0
```

Comment:
Moving this to its own conda-specific section, so it doesn't end up in the `[build-system]` list of requirements. Something like `pip install` / `pip wheel` looks in that list to determine what needs to be installed to build a wheel... if you're using `pip install` / `pip wheel` to do that, then you don't need `build`.